src/routes/cache-hooks/__tests__/event-hooks-queue.e2e-spec.ts

```diff
@@ -0,0 +1,502 @@
+import { amqpClientFactory } from '@/__tests__/amqp-client.factory';
+import { redisClientFactory } from '@/__tests__/redis-client.factory';
+import { retry } from '@/__tests__/util/retry';
+import { TestAppProvider } from '@/__tests__/test-app.provider';
+import { AppModule } from '@/app.module';
+import configuration from '@/config/entities/configuration';
+import { CacheKeyPrefix } from '@/datasources/cache/constants';
+import { CacheDir } from '@/datasources/cache/entities/cache-dir.entity';
+import { faker } from '@faker-js/faker';
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { ChannelWrapper } from 'amqp-connection-manager';
+import { RedisClientType } from 'redis';
+import { getAddress } from 'viem';
+
+describe('Events queue processing e2e tests', () => {
+  let app: INestApplication;
+  let redisClient: RedisClientType;
+  let channel: ChannelWrapper;
+  let queueName: string;
+  const cacheKeyPrefix = crypto.randomUUID();
+  const chainId = '1'; // Mainnet
+
+  beforeAll(async () => {
+    const defaultConfiguration = configuration();
+    const testConfiguration = (): typeof defaultConfiguration => ({
+      ...defaultConfiguration,
+      features: {
+        ...defaultConfiguration.features,
+        eventsQueue: true,
+      },
+    });
+
+    const moduleRef = await Test.createTestingModule({
+      imports: [AppModule.register(testConfiguration)],
+    })
+      .overrideProvider(CacheKeyPrefix)
+      .useValue(cacheKeyPrefix)
+      .compile();
+
+    app = await new TestAppProvider().provide(moduleRef);
+    await app.init();
+    redisClient = await redisClientFactory();
+    const amqpClient = await amqpClientFactory();
+    channel = amqpClient.channel;
+    queueName = amqpClient.queueName;
+  });
+
+  afterAll(async () => {
+    await redisClient.quit();
+    await channel.close();
+    await app.close();
+  });
+
+  it.only.each([
+    {
+      type: 'INCOMING_TOKEN',
+      tokenAddress: faker.finance.ethereumAddress(),
+      txHash: faker.string.hexadecimal({ length: 32 }),
+    },
+    // {
+    //   type: 'OUTGOING_ETHER',
+    //   txHash: faker.string.hexadecimal({ length: 32 }),
+    //   value: faker.string.numeric(),
+    // },
+    // {
+    //   type: 'INCOMING_ETHER',
+    //   txHash: faker.string.hexadecimal({ length: 32 }),
+    //   value: faker.string.numeric(),
+    // },
+    // {
+    //   type: 'OUTGOING_TOKEN',
+    //   tokenAddress: faker.finance.ethereumAddress(),
+    //   txHash: faker.string.hexadecimal({ length: 32 }),
+    // },
```

Comment:
Do we need to keep these here?

Reply:
Absolutely! I commented out these by mistake while debugging 😅 Thanks!

[084c11b](https://github.com/safe-global/safe-client-gateway/pull/1437/commits/084c11b1fc1725728233c9ec9d376916e62a407c)